sway/input/text_input.c

```diff
@@ -597,6 +597,20 @@ static void relay_handle_input_method(struct wl_listener *listener,
 	}
 }
 
+static void relay_handle_text_input_manager_destroy(struct wl_listener *listener, void *data) {
+	struct sway_input_method_relay *relay = wl_container_of(listener, relay,
+		text_input_manager_destroy);
+
+	sway_input_method_relay_finish_input_method(relay);
+}
+
+static void relay_handle_input_method_manager_destroy(struct wl_listener *listener, void *data) {
+	struct sway_input_method_relay *relay = wl_container_of(listener, relay,
+		input_method_manager_destroy);
+
+	sway_input_method_relay_finish(relay);
```

Comment:
Probably we need to do the same here? ie, the input method manager may get destroyed before the text input manager?

Reply:
done in 8e75dad0.